podman: 1.3.2 -> 1.4.4

saschagrunert opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Update podman to the latest version
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

saschagrunert committed:
podman: 1.3.2 -> 1.4.4

Signed-off-by: Sascha Grunert <mail@saschagrunert.de>

saschagrunert commented:
Ah I saw that we have #63353, too. I can rebase on top :)

saschagrunert commented:
@GrahamcOfBorg build podman

bachp reviewed (approved):
Tested and it's working

vdemeester reviewed (approved):
LGTM :tiger: 
I am fine with closing mine :wink:

saschagrunert commented:
> LGTM 
> I am fine with closing mine

Oh that's nice, tanks :pray:

worldofpeace commented:
https://github.com/containers/libpod/releases/tag/v1.4.4 is released.

When you update this PR @saschagrunert  feel free to ping so I can merge it.

saschagrunert commented:
> https://github.com/containers/libpod/releases/tag/v1.4.4 is released.
> 
> When you update this PR @saschagrunert feel free to ping so I can merge it.

Hey @worldofpeace thanks for the hint. I updated it to 1.4.4 and changed the SHA as well as the commit message.

worldofpeace commented:
Get an error when running it without any arguments
```
Error: could not get runtime: could not find a working conmon binary (configured options: [/usr/libexec/podman/conmon /usr/local/lib/podman/conmon /usr/bin/conmon /usr/sbin/conmon /usr/local/bin/conmon /usr/local/sbin/conmon]): invalid argument
```

This conf doesn't look nixos friendly 
* https://github.com/containers/libpod/blob/a22a32a0a5507811168e0251a92122cf3e74ef51/libpod.conf#L8

Should this be something we're concerned with?

saschagrunert commented:
> Get an error when running it without any arguments
> 
> ```
> Error: could not get runtime: could not find a working conmon binary (configured options: [/usr/libexec/podman/conmon /usr/local/lib/podman/conmon /usr/bin/conmon /usr/sbin/conmon /usr/local/bin/conmon /usr/local/sbin/conmon]): invalid argument
> ```
> 
> This conf doesn't look nixos friendly
> 
> * https://github.com/containers/libpod/blob/a22a32a0a5507811168e0251a92122cf3e74ef51/libpod.conf#L8
> 
> Should this be something we're concerned with?

Not right now, since I always have to provide my own config to get things working on nixos. I will follow up on  this topic and will add the current system path there, but I think this is not the only problematic configuration option...

worldofpeace commented:
Huh, yeah rather problematic :rofl: 

I'm not sure why they need to configuration option for paths to executables, but maybe like a build flag would help downstreams.

saschagrunert commented:
> Huh, yeah rather problematic 
> 
> I'm not sure why they need to configuration option for paths to executables, but maybe like a build flag would help downstreams.

I already implemented a path discovery feature for executables in CRI-O, so let's see if we can put it into libpod as well. :)

saschagrunert commented:
Thanks for the fast response :pray:

worldofpeace commented:
> Thanks for the fast response :pray:

Active nixpkgs times for me :smile: .

If you ever think something's good for merge I'm typically responsive.
